test/integration/good/code-gen/standalone_functions/cpp.expected

```diff
@@ -0,0 +1,578 @@
+typename T_lp_accum__>
+void
+test_lp(const T0__& a, T_lp__& lp__, T_lp_accum__& lp_accum__,
+        std::ostream* pstream__) {
+  using local_scalar_t__ = stan::promote_args_t<T0__>;
+  local_scalar_t__ DUMMY_VAR__(std::numeric_limits<double>::quiet_NaN());
+  (void) DUMMY_VAR__;  // suppress unused var warning
+  
+  try {
+    current_statement__ = 15;
+    lp_accum__.add(normal_lpdf<propto__>(a, 0, 1));
+  } catch (const std::exception& e) {
+    stan::lang::rethrow_located(e, locations_array__[current_statement__]);
+      // Next line prevents compiler griping about no return
+      throw std::runtime_error("*** IF YOU SEE THIS, PLEASE REPORT A BUG ***"); 
+  }
+  
+}
+
+struct test_lp_functor__ {
+template <bool propto__, typename T0__, typename T_lp__,
+typename T_lp_accum__>
+void
+operator()(const T0__& a, T_lp__& lp__, T_lp_accum__& lp_accum__,
+           std::ostream* pstream__)  const 
+{
+return test_lp<propto__>(a, lp__, lp_accum__, pstream__);
+}
+};
+
+template <typename T0__, typename RNG>
+stan::promote_args_t<T0__>
+test_rng(const T0__& a, RNG& base_rng__, std::ostream* pstream__) {
+  using local_scalar_t__ = stan::promote_args_t<T0__>;
+  const static bool propto__ = true;
+  (void) propto__;
+  local_scalar_t__ DUMMY_VAR__(std::numeric_limits<double>::quiet_NaN());
+  (void) DUMMY_VAR__;  // suppress unused var warning
+  
+  try {
+    current_statement__ = 17;
+    return normal_rng(a, 1, base_rng__);
+  } catch (const std::exception& e) {
+    stan::lang::rethrow_located(e, locations_array__[current_statement__]);
+      // Next line prevents compiler griping about no return
+      throw std::runtime_error("*** IF YOU SEE THIS, PLEASE REPORT A BUG ***"); 
+  }
+  
+}
+
+struct test_rng_functor__ {
+template <typename T0__, typename RNG>
+stan::promote_args_t<T0__>
+operator()(const T0__& a, RNG& base_rng__, std::ostream* pstream__)  const 
+{
+return test_rng(a, base_rng__, pstream__);
+}
+};
+
+template <bool propto__, typename T0__, typename T1__>
+stan::promote_args_t<T0__,
+T1__>
+test_lpdf(const T0__& a, const T1__& b, std::ostream* pstream__) {
+  using local_scalar_t__ = stan::promote_args_t<T0__, T1__>;
+  local_scalar_t__ DUMMY_VAR__(std::numeric_limits<double>::quiet_NaN());
+  (void) DUMMY_VAR__;  // suppress unused var warning
+  
+  try {
+    current_statement__ = 19;
+    return normal_lpdf<false>(a, b, 1);
+  } catch (const std::exception& e) {
+    stan::lang::rethrow_located(e, locations_array__[current_statement__]);
+      // Next line prevents compiler griping about no return
+      throw std::runtime_error("*** IF YOU SEE THIS, PLEASE REPORT A BUG ***"); 
+  }
+  
+}
+
+struct test_lpdf_functor__ {
+template <bool propto__, typename T0__, typename T1__>
+stan::promote_args_t<T0__,
+T1__>
+operator()(const T0__& a, const T1__& b, std::ostream* pstream__)  const 
+{
+return test_lpdf<propto__>(a, b, pstream__);
+}
+};
+}
+
+// [[Rcpp::export]]
```

Comment:
Is the need for this just to allow Rcpp to not need to use our `basic_model_namespace`? Maybe this is old and @bgoodri  knows? 

If so maybe we can just elide the namespace part for standalone functions and add the Rcpp::export comments to the original definitions...?

Reply:
Do you mean in the sense why don't we just put the stuff in line 52 in `test/integration/good/code-gen/standalone_functions/cpp.expected` outside the namespace?

Reply:
Yeah, and add the `Rcpp::export` comment.

Reply:
That was added due to this comment by Ben: https://github.com/stan-dev/stanc3/pull/524#issuecomment-619211360

Reply:
I was aiming at being close to the old version, but yeah. If Rcpp can handle those templates and all that would make it simpler.

Reply:
The exported functions need to be in the global namespace and not templated, which is why we have been exporting untemplated functions that call the corresponding versions within the namespace.

Reply:
Thanks. In that case we need these non-templated functions and there is no need to remove the namespace from the templated ones.